### ISSUE TITLE
Unable to build vue-data-tables with require or import using Webpack; Element UI works fine

### ISSUE THREAD

**Issue by metamet** [njleonzhang/vue-data-tables#92]
Using Webpack, which has properly built Vue and Element UI using both import and require. Running into issues with vue-data-tables, however.

When using the following: 

```
import DataTables from 'vue-data-tables'
Vue.use(DataTables)
```

I get the error: 

```
vue.js:4400 
Uncaught TypeError: Cannot read property 'install' of undefined
    at Function.Vue.use (vue.js:4400)
    at Object._typeof2 (main.js:16)
    at __webpack_require__ (bootstrap 21bdc8c1cd5b778aa7a2:19)
    at _typeof (bootstrap 21bdc8c1cd5b778aa7a2:65)
    at bootstrap 21bdc8c1cd5b778aa7a2:65
```

From vue.js, here're lines 4388 through 4408 (line 4400 marked with =>):

```
/*  */

function initUse (Vue) {
  Vue.use = function (plugin) {
    var installedPlugins = (this._installedPlugins || (this._installedPlugins = []));
    if (installedPlugins.indexOf(plugin) > -1) {
      return this
    }

    // additional parameters
    var args = toArray(arguments, 1);
    args.unshift(this);
=>if (typeof plugin.install === 'function') {
      plugin.install.apply(plugin, args);
    } else if (typeof plugin === 'function') {
      plugin.apply(null, args);
    }
    installedPlugins.push(plugin);
    return this
  };
}
```

If I change it to: 

```
const DataTables = require('vue-data-tables');
Vue.use(DataTables);
```

I get: 

```
Unknown custom element: <data-tables> - did you register the component correctly? For recursive components, make sure to provide the "name" option.

found in

---> <Anonymous>
       <Root>
```

So if I then declare the named component: 

```
const DataTables = require('vue-data-tables');
Vue.use(DataTables);
Vue.component('data-tables', DataTables);
```

The console error is now: 

```
Failed to mount component: template or render function not defined.

found in

---> <DataTables>
       <Anonymous>
         <Root>
```

Please advise. I'd like to take advantage of the functionality and work you've done, but I simply can't get it to work. Thank you.

**Comment by njleonzhang** [comment]
OK, I will look into this case later ASAP

**Comment by njleonzhang** [comment]
@metamet I have tried in my environment. It can work as expected.
Do you use the lastest version (2.1.0)?
Can you provide a sample project?

**Comment by takotuesday** [comment]
@njleonzhang I actually am running into the same problem. It happens when I build `vue-data-tables` locally and copy the minified `js` file into my assets directory. I wasnt able to use `vue-data-tables` as a package because of an odd error where webpack was looking for `pug`.

**Comment by njleonzhang** [comment]
@takotuesday @metamet I can not reproduce this issue by building the lib locally and then import it. Can you provide a sample project?

All these works:
```
  import DataTables, { DataTablesServer } from '../dist/data-tables.min'
  // import { DataTables, DataTablesServer } from '../dist/data-tables.min'   or in this way

  import Vue from 'vue'

  Vue.use(DataTables)
  Vue.use(DataTablesServer)
```

check the `npm run play` command of this project, it's a sample project.

**Comment by metamet** [comment]
Still getting the same error. I ran `npm install vue-data-tables --save-dev` and have 2.1.0 installed. 

In my main.js, used: 

```
import { DataTables, DataTablesServer } from 'vue-data-tables/dist/data-tables.min'
Vue.use(DataTables)
Vue.use(DataTablesServer)
```

Still getting: 

```
Uncaught TypeError: Cannot read property 'install' of undefined
    at Function.Vue.use (vue.js:4400)
    at Object._typeof2 (main.js:19)
    at __webpack_require__ (bootstrap 4ec7befa776b07ffe73b:19)
    at _typeof2 (bootstrap 4ec7befa776b07ffe73b:65)
    at bootstrap 4ec7befa776b07ffe73b:65
```

What does your `.babelrc` file look like? Mine is: 

```
{
  "presets": ["es2015"],
  "plugins": ["transform-vue-jsx"]
}
```

Here's my `webpack.config.js` as well, in case there's any insight here...

```
var path = require('path')
var webpack = require('webpack')

var entries = {
  Home: './Views/Home/Vue/main.js',
  Requests: './Views/Requests/Vue/main.js',
  Review: './Views/Review/Vue/main.js',
  Shared: './Views/Shared/Vue/main.js'
}

module.exports = {
  entry: entries,
  output: {
    path: path.resolve(__dirname, './wwwroot/Scripts/app'),
    publicPath: '/Scripts/app',
    filename: '[name]/build.js'
  },
  module: {
    rules: [
      { 
        test: /\.html$/, 
        loader: 'vue-html-loader' 
      },
      {
        test: /\.vue$/,
        loader: 'vue-loader',
        options: {
          // vue-loader options go here
        }
      },
      {
        test: /\.(js|jsx)$/,
        loader: 'babel-loader'
      },
      {
        test: /\.(png|jpg|gif|svg)$/,
        loader: 'file-loader',
        options: {
          name: '[name].[ext]?[hash]'
        }
      }
    ]
  },
  resolve: {
    alias: {
      'vue$': 'vue/dist/vue.js'
    }
  },
  devServer: {
    historyApiFallback: true,
    noInfo: true
  },
  devtool: '#eval-source-map'
}

if (process.env.NODE_ENV === 'production') {
  module.exports.devtool = '#source-map'
  // http://vue-loader.vuejs.org/en/workflow/production.html
  module.exports.plugins = (module.exports.plugins || []).concat([
    new webpack.DefinePlugin({
      'process.env': {
        NODE_ENV: '"production"'
      }
    }),
    new webpack.LoaderOptionsPlugin({
      minimize: true
    })
  ])
}
```

Thanks for looking into this, and sorry for my delayed response.

**Comment by njleonzhang** [comment]
@metamet I have not find any suspicious point in code you pasted above. I create a project from vue-webapck template, [vue-data-table-issue-92](https://github.com/njleonzhang/vue-data-table-issue-92), which can use vue-data-tables as expected. Can you check it, and try to find the difference to yours. You can also provide a sample project so that we can help to debug.

**Comment by johncade** [comment]
Something else recently happened for me when using vue-data-tables in production. Any time I had a custom column template in a data-table, for example like:
```html
<el-column value="name" label="Name">
    <template slot-scope="scope">
     <span> {{ scope.row.name.toUpperCase() }} </span>
    </template>
</el-column>
```
I get an error along the likes of cannot get `row` of `undefined`. This only happened when using `vue-data-tables`. When I switched the `vue-data-table` wrapper to `el-table` the errors went away. The odd thing is, this only happened when I deployed to Heroku, and not locally. I believe it has to do with these inconsistencies involved with webpack because I have yet been able to reproduce it.
